### PR TITLE
Combined contribute_to_qiskit.rst with contributing.md

### PR DESCRIPTION
**Pull request by javabster**
### Summary

This PR copies over relevant information in `contribute_to_qiskit.rst` into the `CONTRIBUTING.md`, meaning we can safely remove https://qiskit.org/documentation/contributing_to_qiskit.html when the time comes.

This PR also updates `CONTRIBUTING.md` to no longer use the term "terra"

### CODE REVIEW

File: CONTRIBUTING.md
Comment (Eric-Arellano) [inline CONTRIBUTING.md]: Nit: let's lower expectations hah

```suggestion
* [good first issue](https://github.com/Qiskit/qiskit/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) - these issues are typically the simplest available to work on, ideal for newcomers. They should already be fully scoped, with a clear approach outlined in the descriptions.
```

File: CONTRIBUTING.md
Comment (Eric-Arellano) [inline CONTRIBUTING.md:103]: Would it make sense to instead link out to the new Install From Source page in OneDocs? I think the main motivation would be de-duplication.

I'm fine with it all being in this guide, though.

**Review by Eric-Arellano — DISMISSED** [review]
Thanks, Abby!

File: CONTRIBUTING.md
Comment (javabster) [reply CONTRIBUTING.md:103]: I'm happy with that, but maybe the maintainers have a different opinion? this would be the new link if we decided to change: https://docs.quantum-computing.ibm.com/start/install-qiskit-source

Comment (javabster) [reply CONTRIBUTING.md:103]: The content is a bit different, so maybe theres reason to keep both?

**Comment by Eric-Arellano** [comment]
Bump @Qiskit/terra-core

File: CONTRIBUTING.md
Comment (jakelishman) [inline CONTRIBUTING.md]: The refactoring here seems to have broken the logical heading levels a little bit.  Not sure what heading structure you prefer, but at the moment we go from level 1 to level 3 then back to level 2, which isn't great for accessibility.

File: CONTRIBUTING.md
Comment (jakelishman) [inline CONTRIBUTING.md]:
```suggestion
* Read the [Code of Conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md)
```

File: CONTRIBUTING.md
Comment (jakelishman) [inline CONTRIBUTING.md:184]: In step 2, just above in the folded bits of the diff, there's a link to `qiskit.org/docuemntation/contributing_to_qiskit.html#documentation-structure`, which I think might need changing?

File: CONTRIBUTING.md
Comment (jakelishman) [inline CONTRIBUTING.md]:
```suggestion
[the Qiskit Code of Conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md).
```

**Review by jakelishman — COMMENTED** [review]
Please can we also make sure that the now-obsolete file is deleted from the repository as part of this PR?

File: CONTRIBUTING.md
Comment (javabster) [reply CONTRIBUTING.md]: ah good point, the heading levels where a bit janky to beign with so I tried to tidy them up a bit, but must have missed this one

File: CONTRIBUTING.md
Comment (javabster) [reply CONTRIBUTING.md:184]: looks like that link doesn't even exist in our current qiskit.org page, i'm guessing it was from an old old version of the contributing page. I think safest to just remove

**Review by jakelishman — APPROVED** [review]
Thanks Abby, I'm fine with this.  Given Eric's concerns about maybe needing to make an emergency docs deployment, perhaps the best strategy is to leave this PR approved but not merged until we're ready on the qiskit.org side (or wherever it's getting deployed)?  Having this PR with the old-file removal means git will force us to resolve the change if anybody happens to make a change to the old form before we actually merge this, so we can't accidentally lose changes.

**Comment by jakelishman** [comment]
Apologies - I misremembered: they were your concerns, not Eric's.

**Comment by javabster** [comment]
It's only 1 week until we get the redirects going and I highly doubt we'll need to do any docs deployments at this later stage, so I'm happy to take that risk and deploy this now :)

**Pull request merged**
